### PR TITLE
add priority controller infrastructure

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

File: src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+using LT.DigitalOffice.DashboardService.Validation.Priority.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.DashboardService.Validation.Priority;
+
+public class CreatePriorityRequestValidator : AbstractValidator<CreatePriorityRequest>, ICreatePriorityRequestValidator
+{
+
```
Comment (VLiamin) [inline src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs]: empty line

File: src/DashboardService/Controllers/PrioritiesController.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public async Task<OperationResultResponse<Guid?>> CreateAsync(
+    [FromBody] CreatePriorityRequest request,
+    [FromServices] CreatePriorityCommand command)
+  {
+    return await command.ExecuteAsync(request);
+  }
+  
+  public async Task<FindResultResponse<PriorityInfo>> GetAsync(
+    [FromServices] GetPrioritiesCommand command
+    )
```
Comment (VLiamin) [inline src/DashboardService/Controllers/PrioritiesController.cs]: On previous line

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace LT.DigitalOffice.DashboardService.Data.Interfaces;
+
+[AutoInject]
+public interface IPriorityRepository
+{
+  Task<Guid?> CreateAsync(DbPriority dbDepartment);
+  Task<List<DbPriority>> GetAllAsync();
+  Task<DbPriority> GetAsync(Guid id);
+  Task<bool> EditAsync(Guid id);
```
Comment (AnnPostnikova) [inline src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: patch request was lost

File: src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs (hunk 1 above)
Comment (Glunky) [reply src/DashboardService.Validation/Priority/CreatePriorityRequestValidator.cs]: ++

File: src/DashboardService/Controllers/PrioritiesController.cs (hunk 2 above)
Comment (Glunky) [reply src/DashboardService/Controllers/PrioritiesController.cs]: ++

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs (hunk 3 above)
Comment (Glunky) [reply src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: editAsync will be used for patch request 
image

Comment (AnnPostnikova) [reply src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: Based on what model will changes be made to the repository? Only the id is sent to the repository

File: src/DashboardService/Controllers/PrioritiesController.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    [FromBody] JsonPatchDocument<PatchPriorityRequest> request,
+    [FromServices] EditPriorityCommand command)
+  {
+    return await command.ExecuteAsync(id, request);
+  }
+  
+  [HttpDelete("{id}")]
+  public async Task<OperationResultResponse<bool>> RemoveAsync(
+    [FromRoute] Guid id,
+    [FromServices] RemovePriorityCommand command)
```
Comment (VLiamin) [inline src/DashboardService/Controllers/PrioritiesController.cs]: I do not understand in which rule you place  [FromRoute], [FromServices] and [FromBody]. At each endpoint they change their location

Comment (Glunky) [reply src/DashboardService/Controllers/PrioritiesController.cs]: ++

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs (hunk 3 above)
Comment (Glunky) [reply src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: ++

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Data.Interfaces;
+
+[AutoInject]
+public interface IPriorityRepository
+{
+  Task<Guid?> CreateAsync(DbPriority dbDepartment);
+  Task<List<DbPriority>> GetAllAsync();
```
Comment (SolovyovDmitry) [inline src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: you will pass filter

File: src/DashboardService.Models.Dto/Requests/Priority/PatchPriorityRequest.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+using System;
+
+namespace LT.DigitalOffice.DashboardService.Models.Dto.Requests.Priority;
+
+public record PatchPriorityRequest
+{
+  public string PriorityName { get; set; }
```
Comment (SolovyovDmitry) [inline src/DashboardService.Models.Dto/Requests/Priority/PatchPriorityRequest.cs]: just name

**Review by SolovyovDmitry — APPROVED** [review]

File: src/DashboardService.Models.Dto/Requests/Priority/PatchPriorityRequest.cs (hunk 6 above)
Comment (Glunky) [reply src/DashboardService.Models.Dto/Requests/Priority/PatchPriorityRequest.cs]: ++

File: src/DashboardService.Data/Interfaces/IPriorityRepository.cs (hunk 5 above)
Comment (Glunky) [reply src/DashboardService.Data/Interfaces/IPriorityRepository.cs]: ++

File: src/DashboardService/Controllers/PrioritiesController.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  [HttpGet("{id}")]
+  public async Task<OperationResultResponse<PriorityInfo>> GetAsync(
+    [FromServices] GetPriorityCommand command,
+    [FromRoute] Guid id)
+  {
+    return await command.ExecuteAsync(id);
+  }
+
+  [HttpPatch("{id}")]
+  public async Task<OperationResultResponse<bool>> PatchAsync(
```
Comment (VLiamin) [inline src/DashboardService/Controllers/PrioritiesController.cs]: EditAsync

**Review by VLiamin — APPROVED** [review]

**Pull request merged**